File: day-01/part-1/youyoun.rb
def is_number?(obj)
    return obj.ord >= 48 && obj.ord <= 57
end

def run(s)
    sum = 0
    for line in s
        first, last = nil, nil
        first_idx = nil
        line.each_char do |c|
            if is_number?(c)
                if first.nil?
                    first = c.to_i
                end
                last = c.to_i
            end
        end
        sum += first * 10 + last
    end
    return sum
end


starting = Process.clock_gettime(Process::CLOCK_MONOTONIC)
answer = run(ARGV[0].lines)
elapsed = (Process.clock_gettime(Process::CLOCK_MONOTONIC) - starting) * 1000

puts "_duration:#{elapsed}\n#{answer}"

File: day-01/part-1/youyoun.rs
use std::env::args;
use std::time::Instant;

fn main() {
    let now = Instant::now();
    let output = run(&args().nth(1).expect("Please provide an input"));
    let elapsed = now.elapsed();
    println!("_duration:{}", elapsed.as_secs_f64() * 1000.);
    println!("{}", output);
}

fn run(input: &str) -> u32 {
    // Your code goes here
    let mut s: u32 = 0;
    for line in input.lines() {
        let mut is_first_set: bool = false;
        let mut first: u32 = 0;
        let mut last: u32 = 0;
        for c in line.chars() {
            if c.is_ascii_digit() {
                let digit: u32 = c.to_digit(10).unwrap();

                if !is_first_set {
                    first = digit;
                    is_first_set = true;
                }
                last = digit;
            }
        }
        s += first * 10 + last;
    }
    s
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn run_test() {
        assert_eq!(run("Test example"), 0)
    }
}
